File: problem_statement_1/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {

    // OPEN THE FILE
    FILE *file = fopen("problem_statement_1/numbers.txt", "r");

    if (file == NULL) {
        printf("Error opening file\n");
        return 1;
    }

    char number[20];

    // READ EACH NUMBER FROM FILE
    while (fscanf(file, "%s", number) != EOF) {

        // CHECK LENGTH = 6
        if (strlen(number) != 6) {
            continue;
        }

        int valid = 1;

        // CHECK EACH DIGIT ONE BY ONE

        // O O O E E O

        // 1st digit (odd)
        if ((number[0] - '0') % 2 == 0) {
            valid = 0;
        }

        // 2nd digit (odd)
        if ((number[1] - '0') % 2 == 0) {
            valid = 0;
        }

        // 3rd digit (odd)
        if ((number[2] - '0') % 2 == 0) {
            valid = 0;
        }

        // 4th digit (even)
        if ((number[3] - '0') % 2 != 0) {
            valid = 0;
        }

        // 5th digit (even)
        if ((number[4] - '0') % 2 != 0) {
            valid = 0;
        }

        // 6th digit (odd)
        if ((number[5] - '0') % 2 == 0) {
            valid = 0;
        }

        // IF PATTERN FAILED → SKIP
        if (valid == 0) {
            continue;
        }

        // CONVERT STRING TO INTEGER
        int num = atoi(number);

        // CHECK DIVISIBLE BY 13
        if (num % 13 == 0) {
            printf("%s\n", number);
        }
    }

    fclose(file);
    return 0;
}

File: problem_statement_1/main.py
# FIRST OF ALL, IMPORT THE NUMBERS FILE:

with open("problem_statement_1/numbers.txt") as file:
    # READ THE CONTENT OF THE FILE, SPLIT
    numbers = file.read().split()

    # FILTER ONLY 6 DIGIT NUMBERS
    #CHECK USING LEN OF STRING REPRESENTATION OF THE NUMBER
    six_digit_numbers = []
    for number in numbers:
        if len(number) == 6:
            six_digit_numbers.append(number)

    # CHECK FOR PATTERN OOOEEO
    pattern_numbers = []
    for number in six_digit_numbers:
        #SINCE WE CAN USE STRING INDEXING TO ACCESS EACH DIGIT, WE CAN CHECK THE PATTERN DIRECTLY
        #WE CONVERT THE CHARACTER TO INTEGER TO PERFORM MODULO OPERATION
        if (int(number[0]) % 2 != 0 and   # O 
            int(number[1]) % 2 != 0 and   # O 
            int(number[2]) % 2 != 0 and   # O
            int(number[3]) % 2 == 0 and   # E
            int(number[4]) % 2 == 0 and   # E
            int(number[5]) % 2 != 0):     # O
            pattern_numbers.append(number)
    
    # NOW CHECK IF NUMBER IS DIVISIBLE BY 13
    divisible_by_13 = []
    for number in pattern_numbers:
        if int(number) % 13 == 0:
            divisible_by_13.append(number)

    print(divisible_by_13)